core/src/test/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingServiceSpec.scala

```diff
@@ -166,14 +281,177 @@ class SpendReportingServiceSpec extends AnyFlatSpecLike with TestDriverComponent
+            withClue(s"cost for ${SpendReportingTestData.SubAggregation.workspace1.toWorkspaceName} was incorrect") {
+              spendData.cost shouldBe SpendReportingTestData.SubAggregation.workspace1TotalCostRounded.toString
+            }
+            subAggregation.spendData.map { subAggregatedSpendData =>
+              if (subAggregatedSpendData.category == Option(TerraSpendCategories.Compute)) {
+                withClue(s"${subAggregatedSpendData.category} category cost for ${SpendReportingTestData.SubAggregation.workspace1.toWorkspaceName} was incorrect") {
+                  subAggregatedSpendData.cost shouldBe SpendReportingTestData.SubAggregation.workspace1ComputeRowCostRounded.toString
+                }
+              } else if (subAggregatedSpendData.category == Option(TerraSpendCategories.Other)) {
+                withClue(s"${subAggregatedSpendData.category} category cost for ${SpendReportingTestData.SubAggregation.workspace1.toWorkspaceName} was incorrect") {
+                  subAggregatedSpendData.cost shouldBe SpendReportingTestData.SubAggregation.workspace1OtherRowCostRounded.toString
+                }
+              } else {
+                fail(s"unexpected category found in spend results - $subAggregatedSpendData")
+              }
+            }
+
+          } else if (workspaceGoogleProject.equals(SpendReportingTestData.SubAggregation.workspace2.googleProjectId.value)) {
+            withClue(s"cost for ${SpendReportingTestData.SubAggregation.workspace2.toWorkspaceName} was incorrect") {
+              spendData.cost shouldBe SpendReportingTestData.SubAggregation.workspace2TotalCostRounded.toString
+            }
+            subAggregation.spendData.map { subAggregatedSpendData =>
+              if (subAggregatedSpendData.category == Option(TerraSpendCategories.Storage)) {
+                withClue(s"${subAggregatedSpendData.category} category cost for ${SpendReportingTestData.SubAggregation.workspace2.toWorkspaceName} was incorrect") {
+                  subAggregatedSpendData.cost shouldBe SpendReportingTestData.SubAggregation.workspace2StorageRowCostRounded.toString
+                }
+              } else if (subAggregatedSpendData.category == Option(TerraSpendCategories.Other)) {
+                withClue(s"${subAggregatedSpendData.category} category cost for ${SpendReportingTestData.SubAggregation.workspace2.toWorkspaceName} was incorrect") {
+                  subAggregatedSpendData.cost shouldBe SpendReportingTestData.SubAggregation.workspace2OtherRowCostRounded.toString
+                }
+              } else {
+                fail(s"unexpected category found in spend results - $subAggregatedSpendData")
+              }
+            }
+          } else {
+            fail(s"unexpected workspace found in spend results - $spendData")
+          }
+        }
+      }
+      case SpendReportingAggregation(SpendReportingAggregationKeys.Category, spendData) => {
```

Comment:
I this this block of code is the same as an earlier test, where you verified the ability to aggregate by Category. Similar to my last comment, could you share the code?